app/web/features/auth/notifications/NotificationSettings.tsx

```diff
@@ -0,0 +1,87 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import { notificationSettingsQueryKey } from "features/queryKeys";
+import { Error as GrpcError } from "grpc-web";
+import {
+  GetNotificationSettingsRes,
+  SetNotificationSettingsRes,
+} from "proto/notifications_pb";
+import { useTranslation } from "react-i18next";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { service } from "service";
+
+interface NotificationSettingFormData {
+  newNotificationsEnabled: boolean;
+}
+
+export default function NotificationSettings({
+  className,
+}: {
+  className: string;
+}) {
+  const { t } = useTranslation("auth");
+
+  const queryClient = useQueryClient();
+
+  const { data, error, isLoading } = useQuery<
+    GetNotificationSettingsRes.AsObject,
+    GrpcError
+  >(
+    notificationSettingsQueryKey,
+    service.notifications.getNotificationSettings
+  );
+
+  const mutation = useMutation<
+    SetNotificationSettingsRes.AsObject,
+    GrpcError,
+    NotificationSettingFormData
+  >(
+    ({ newNotificationsEnabled }) =>
+      service.notifications.setNotificationSettings(newNotificationsEnabled),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(notificationSettingsQueryKey);
+      },
+    }
+  );
+
+  const toggleNewNotifications = async () => {
+    mutation.mutate({
+      newNotificationsEnabled: !data?.newNotificationsEnabled,
+    });
+  };
```

Comment:
```suggestion
  const toggleNewNotifications = async () => {
    if (!data) return;
    mutation.mutate({
      newNotificationsEnabled: !data.newNotificationsEnabled,
    });
  };
```

Alternatively this. It invisibly does nothing if data is undefined, but if that state is impossible (ie. this function is never called because the button isn't shown when data is undefined), maybe it doesn't matter

Reply:
I did this, thanks!